Sources/SwiftDiagnostics/GroupedDiagnostics.swift

```diff
@@ -133,6 +133,32 @@ extension GroupedDiagnostics {
     }
   }
 
+  // Find the "primary" diagnostic that will be shown at the top of the diagnostic
+  // message. This is typically the error, warning, or remark.
+  private func findPrimaryDiagnostic(in sourceFile: SourceFile) -> (SourceFile, Diagnostic)? {
+    // If there is a non-note diagnostic, it's the primary diagnostic.
+    if let primaryDiag = sourceFile.diagnostics.first(where: { $0.diagMessage.severity != .note }) {
```

Comment:
Should we also check for remark here? Would it make sense that `Severity` is `Comparable` for that?

Reply:
Remarks are treated like warnings or errors, and can have notes attached.